src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -397,8 +397,11 @@ public function getWriteInfo(string $class, string $property, array $context = [
 
         if ($reflClass->hasProperty($property) && ($reflClass->getProperty($property)->getModifiers() & $this->propertyReflectionFlags)) {
             $reflProperty = $reflClass->getProperty($property);
+            if (\PHP_VERSION_ID < 80100 || (false === $reflProperty->isReadOnly() && false === $reflProperty->isPromoted())) {
```

Comment:
If I remove the part about `$reflProperty->isPromoted()` (which I just did and pushed), tests still pass. This makes me wonder what how this relates to property promotion? Isn't this only about readonly?

Reply:
That's my mistake. I've missed the part that readonly properties in the constructor are automatically promoted. The change above indeed is correct.